### PR TITLE
Using the localized texts in UI 

### LINKED ISSUE #855: Some UI text is not properly localized
The following UI text is not properly localized.

Peek-Definition
Execute Query
Messages
Total Execution time
Save as Json, CSV …
Code snippet descriptions “Select Rows from a table or a view”

### PR DESCRIPTION
**Pull request by llali**
- Added the texts used in views to localizedConstants.enu.xlf to get localized.
- Loading the texts in Views/Constants from the localized ones before using in UI
- Making extension commands and config localizable 
Fixed #855 and #783

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -337,93 +337,93 @@
...
-                "description": "[Optional] Indicates whether this profile has an empty password explicitly set."
+                "description": "%mssql.connection.emptyPasswordInput%"
               }
             }
           }
         },
         "mssql.shortcuts": {
           "type": "object",
-          "description": "Shortcuts related to the results window",
+          "description": "%mssql.shortcut%",
```
Comment (llali) [inline package.json]: >mssql.shortcut [](start = 27, length = 14)

will fix this. should be %mssql.shortcuts%

File: src/models/sqlOutputContentProvider.ts [hunk 2]
```diff
@@ -66,6 +66,14 @@ export class SqlOutputContentProvider implements vscode.TextDocumentContentProvi
         this._service.addPostHandler(Interfaces.ContentType.ShowError, (req, res) => this.showErrorRequestHandler(req, res));
         // add http post handler for showing warning to user
         this._service.addPostHandler(Interfaces.ContentType.ShowWarning, (req, res) => this.showWarningRequestHandler(req, res));
+        this._service.addHandler(Interfaces.ContentType.LocalizedText, (req, res) => {
```
Comment (llali) [inline src/models/sqlOutputContentProvider.ts]: will add comment

File: package.nls.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{
+  "extension.runQuery" : "Execute Query",
```
Comment (kevcunnane) [inline package.nls.json]: Is there an xliff file for these? And a process to build the localized package.nls files during build?

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 4]
```diff
@@ -321,6 +321,14 @@ export class AppComponent implements OnInit, AfterViewChecked {
         const self = this;
         this.setupResizeBind();
 
+        Constants.ConstantTypes.forEach(element => {
```
Comment (kevcunnane) [inline src/views/htmlcontent/src/js/components/app.component.ts]: Please do a single call and return all constants.

File: src/views/htmlcontent/src/js/constants.ts [hunk 5]
```diff
@@ -1,19 +1,57 @@
...
 
 /** Warning message for save icons */
-export const msgCannotSaveMultipleSelections = 'Save results command cannot be used with multiple selections.';
+export let msgCannotSaveMultipleSelections = 'Save results command cannot be used with multiple selections.';
+
+export let loadLocalizedConstant = (key: string, value: string) => {
+    this[key] = value;
+};
+
+export enum ConstantType {
```
Comment (kevcunnane) [inline src/views/htmlcontent/src/js/constants.ts]: You can get rid of this once it's a map. thanks!

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 6]
```diff
@@ -321,6 +321,7 @@ export class AppComponent implements OnInit, AfterViewChecked {
         const self = this;
         this.setupResizeBind();
 
+
```
Comment (kevcunnane) [inline src/views/htmlcontent/src/js/components/app.component.ts]: Minor: remove this whitespace

File: src/views/htmlcontent/src/js/constants.ts [hunk 7]
```diff
@@ -1,19 +1,24 @@
...
+export let messagePaneLabel = 'Messages';
+export let lineSelectorFormatted = 'Line {0}';
+export let elapsedTimeLabel = 'Total execution time: {0}';
 
 /** Warning message for save icons */
-export const msgCannotSaveMultipleSelections = 'Save results command cannot be used with multiple selections.';
+export let msgCannotSaveMultipleSelections = 'Save results command cannot be used with multiple selections.';
+
+export let loadLocalizedConstant = (key: string, value: string) => {
+    this[key] = value;
```
Comment (kevcunnane) [inline src/views/htmlcontent/src/js/constants.ts]: Minor: maybe document what this does / how it works. It's kinda magic (but cool)

File: src/views/htmlcontent/src/js/constants.ts [hunk 8]
```diff
@@ -1,19 +1,24 @@
...
-export const maximizeLabel = 'Maximize';
-export const restoreLabel = 'Restore';
-export const saveCSVLabel = 'Save as CSV';
-export const saveJSONLabel = 'Save as JSON';
-export const saveExcelLabel = 'Save as Excel';
-export const resultPaneLabel = 'Results';
-export const selectAll = 'Select all';
-export const copyLabel = 'Copy';
-export const copyWithHeadersLabel = 'Copy with Headers';
+export let maximizeLabel = 'Maximize';
```
Comment (kevcunnane) [inline src/views/htmlcontent/src/js/constants.ts]: Please doc that all new strings added here must be added to the xliff. Otherwise people may miss this

**Pull request merged**